### PR TITLE
ApiManagement Resource Schema for 2017-03-01

### PR DESCRIPTION
**Pull request by solankisamir**

### CODE REVIEW

**Comment by solankisamir** [comment]
@TianoMS I am getting this error, trying to validate the spec

> C:\github\azure-resource-manager-schemas\tools>node validateJSON.js ..\schemas\2017-03-01\Microsoft.ApiManagement.json RsourceMetaSchema.json ..\schemas\
module.js:471
    throw err;
    ^

Error: Cannot find module 'tv4'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\github\azure-resource-manager-schemas\tools\validateJSON.js:5:13)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)

**Comment by TianoMS** [comment]
I guess you need to run "npm install" to install the modules.

**Comment by solankisamir** [comment]
Json Validation Passed

File: schemas/2017-03-01/Microsoft.ApiManagement.json [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+          ],
+          "description": "Sku properties of the Api Management service."
+        }
+      },
+      "required": [
+        "type",
+        "apiVersion",
+        "location",
+        "properties",
+        "sku"
```
Comment (TianoMS) [inline schemas/2017-03-01/Microsoft.ApiManagement.json]: >sku [](start = 9, length = 3)

Is sku required?

File: schemas/2017-03-01/Microsoft.ApiManagement.json [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+          },
+          "enum": [
+            "Developer",
+            "Standard",
+            "Premium"
+          ],
+          "x-ms-enum": {
+            "name": "SkuType",
+            "modelAsString": false
+          }
```
Comment (TianoMS) [inline schemas/2017-03-01/Microsoft.ApiManagement.json]: are these generated by AutoRest?

File: tests/2017-03-01/Microsoft.ApiManagement.tests.json [hunk 3]
```diff
@@ -0,0 +1,80 @@
+{
+	"tests": [{
```
Comment (TianoMS) [inline tests/2017-03-01/Microsoft.ApiManagement.tests.json]: >	 [](start = 0, length = 1)

Please format. use two spaces as indent.

Comment (solankisamir) [inline tests/2017-03-01/Microsoft.ApiManagement.tests.json]: fixed

**Comment by TianoMS** [comment]
>     "service_apis_policies": {

Put these resource types in deploymentTemplate too?

---
Refers to: schemas/2017-03-01/Microsoft.ApiManagement.json:300 in eec257f. [](commit_id = eec257ff7fe4d4c64c027daf2ad01a5ed9c2ffd7, deletion_comment = False)

File: schemas/2017-03-01/Microsoft.ApiManagement.json (hunk 2 above)
Comment (solankisamir) [inline schemas/2017-03-01/Microsoft.ApiManagement.json]: generated using autorest.

**Comment by TianoMS** [comment]
>               "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"

this looks like a change in AutoRest. Today the deploymentTemplate file ID is "[http://...](http://...)". 

---
Refers to: schemas/2017-03-01/Microsoft.ApiManagement.json:17 in eec257f. [](commit_id = eec257ff7fe4d4c64c027daf2ad01a5ed9c2ffd7, deletion_comment = False)

**Comment by solankisamir** [comment]
>     "service_apis_policies": {

already there.

---
In reply to: [315484514](https://github.com/Azure/azure-resource-manager-schemas/pull/329#issuecomment-315484514) [](ancestors = 315484514)

---
Refers to: schemas/2017-03-01/Microsoft.ApiManagement.json:300 in eec257f. [](commit_id = eec257ff7fe4d4c64c027daf2ad01a5ed9c2ffd7, deletion_comment = False)

File: tests/2017-03-01/Microsoft.ApiManagement.tests.json [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	"tests": [
+		{
+			"name": "ApiManagement service - Minimal",
+			"definition": "http://schema.management.azure.com/schemas/2017-03-01/Microsoft.ApiManagement.json#/resourceDefinitions/service",
+			"json": {
+				"type": "Microsoft.ApiManagement/service",
+				"apiVersion": "2017-03-01",
+				"location": "West Us",
+				"sku": {
+					"name": "Developer",
```
Comment (TianoMS) [inline tests/2017-03-01/Microsoft.ApiManagement.tests.json]: >					 [](start = 0, length = 5)

Seems the indents are not fixed.

File: tests/2017-03-01/Microsoft.ApiManagement.tests.json [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+			"name": "ApiManagement service - Missing Properties",
+			"expectedErrors": [
+				{
+					"message": "Missing required property: properties",
+					"dataPath": "/"
+				}
+			],
+			"definition": "http://schema.management.azure.com/schemas/2017-03-01/Microsoft.ApiManagement.json#/resourceDefinitions/service",
+			"json": {
+				"type": "Microsoft.ApiManagement/service",
```
Comment (TianoMS) [inline tests/2017-03-01/Microsoft.ApiManagement.tests.json]: >service [](start = 37, length = 7)

Add test cases for the other resource types?

**Comment by TianoMS** [comment]
>               ]

Isn't this a problem?  the enum is a string "1" which is not interger. 

---
Refers to: schemas/2017-03-01/Microsoft.ApiManagement.json:1451 in f94b83e. [](commit_id = f94b83e86857c98daa7218bb205ed31011e4f562, deletion_comment = False)

**Pull request merged**